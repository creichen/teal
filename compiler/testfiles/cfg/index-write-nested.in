fun f(a, b, c) = {
  a[b[0] and c] := 2;
}

