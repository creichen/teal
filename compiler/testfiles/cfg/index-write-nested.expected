Entry#0 @ 1 :
  -> IntConstant#1
VarDecl#0 @ 1 :
VarDecl#1 @ 1 :
VarDecl#2 @ 1 :
Access#0 @ 2 :
  -> Access#1
IndexExpr#0 @ 2 :
  -> AssignStmt#0
AssignStmt#0 @ 2 :
  -> Exit#0
Access#1 @ 2 :
  -> IntConstant#0
IndexExpr#1 @ 2 :
  -> Access#2
AndExpr#0 @ 2 :
  -> IndexExpr#0
IntConstant#0 @ 2 :
  -> IndexExpr#1
Access#2 @ 2 :
  -> AndExpr#0
IntConstant#1 @ 2 :
  -> Access#0
Exit#0 @ 3 :
